### DISCUSSION TITLE
Issues with @Inject in test classes while using @IfBuildProperty

### DISCUSSION THREAD

**Discussion by kishorepgopan** [quarkusio/quarkus#31365] [Q&A]
Hi All,
I am trying to create a jar with some utility classes which can be used in multiple services. I don't want to create all the beans and want to give individual apps flexibility to enable the bean creation using properties. So I went ahead to create the beans using `@Produces `and `@IfBuildProperty`. The bean injection is working fine but during the build, the test classes are failing due to `Suppressed: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type com.test.utils.CommonUtils and qualifiers [@Default]`

```

  @Produces
  @IfBuildProperty(name = "utility.enable", stringValue = "true")
  public CommonUtils commonUtils() {
    return new CommonUtils();
  }
```

I have tried the below approaches. None of them worked so far

1.  Used `@ApplicationScoped` +  `@IfBuildProperty`. Properties are added to the application.properties under the test folder resources
2. Created two Config classes in the main folder. One with `@UnlessBuildProfile("test")` along with `@IfBuildProperty` and another with `@IfBuildProfile("test")` without `@IfBuildProperty`
3. Created two config classes, one in the main folder and another in the test folder. The one config inside the test folder doesn't have `@IfBuildProperty`

Can someone suggest how to resolve this issue and what is the best way to create conditional beans if my current approach is not feasible?

**Comment by radcortez** [comment]
Any chance you can provide us with a reproducer so we can look? Thank you!

**Reply by radcortez** [reply]
Closing because we never got a reproducer.
